### PR TITLE
Fix hiding a maximized window On Windows

### LINKED ISSUE #2233: On Windows, hiding a maximized window, unmaximizes instead
### Steps to reproduce:
1. `cargo run --example window_debug`
2. press <kbd>x</kbd> to maximize the window
3. press <kbd>v</kbd> to hide it

### Bug:
The window isn't hidden and isn't responsive. The window actually restored its original unmaximized size instead. The window icon is removed from the taskbar.

### Expected:
The window is hidden, the window icon is removed from the taskbar and the window doesn't restore to its original unmaximized size.

### PR DESCRIPTION
**Pull request by amrbashir**
Fixes #2233

- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

I couldn't find a better way to fix this so I left a comment.

### CODE REVIEW

**Comment by amrbashir** [comment]
maybe a better way to go about this would be to introduce a new method `WindowFlags::diff_mask` and add a comment that it should only be called on the diff between `self` and `new`?

**Comment by msiglreith** [comment]
Thanks but I'm not totally happy with introducing more conditionals.
The issue seems to be that the second `ShowWindow(SW_MAXIMIZE)` call shadows the `SW_HIDE` from earlier due to implicit window activation from `SW_MAXIMIZE`. I currently would prefer to:
- Removing the `VISIBILE` masking in `WindowState::mask`
- Splitting the `ShowWindow` to only the `VISIBLE` case on start of the function and move the `!VISIBLE` case down to the end of the function.

**Comment by amrbashir** [comment]
Thank you for the suggestions, it is much better to handle it that way.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ And please only add new entries to the top of this list, right below the `# Unre
 
 # Unreleased
 
+- On Windows, Fix hiding a maximized window.
```
Comment (filnet) [inline CHANGELOG.md:11]: `Fix` should not be capitalized.

**Review by msiglreith — APPROVED** [review]

**Pull request merged**